quartz_solar_forecast/data.py

```diff
@@ -179,33 +179,48 @@ def process_pv_data(live_generation_kw: Optional[pd.DataFrame], ts: pd.Timestamp
 
     return da
 
-def make_pv_data(site: PVSite, ts: pd.Timestamp) -> xr.Dataset:
+def make_pv_data(
```

Comment:
Linked to my previous comment not sure about having all of these as input parameters vs some of them being created in the function like `givenergy_data`

Reply:
> Linked to my previous comment not sure about having all of these as input parameters vs some of them being created in the function like `givenergy_data`

thats the reason we're first checking in every if inverter_type=="xyz" condition whether the data already exists or not, that already existing data is the prefetched one from the api when the api is used, but when running from the cli, that data would be None, so we run the func to fetch it from the respective inverter file

Reply:
would it be better to just set the enphase token, and then you can sue `make_pv_data` to get the data. You might need some documentation, or some direction how people get that token

Reply:
> would it be better to just set the enphase token, and then you can sue `make_pv_data` to get the data. You might need some documentation, or some direction how people get that token

the enphase token cant be set, it has an additional level of authentication wwhich requires the home owner to approve - which will give the user the access token on demand

Reply:
Ok, im just looking here

https://github.com/openclimatefix/Open-Source-Quartz-Solar-Forecast/blob/main/quartz_solar_forecast/inverters/enphase.py#L52

Currently 
1. get enphase auth url
2. get_enphase_authorization_code (which currently uses CLI)
3. get access token using the `auth code`

Couldnt there be a slightly seperate process to get the `auth code`, then you just set the `auth code` in your env vars and you can then automatically get the access token

Reply:
> Ok, im just looking here
> 
> https://github.com/openclimatefix/Open-Source-Quartz-Solar-Forecast/blob/main/quartz_solar_forecast/inverters/enphase.py#L52
> 
> Currently
> 
> 1. get enphase auth url
> 2. get_enphase_authorization_code (which currently uses CLI)
> 3. get access token using the `auth code`
> 
> Couldnt there be a slightly seperate process to get the `auth code`, then you just set the `auth code` in your env vars and you can then automatically get the access token

the thing with auth code isthat it keeps changing , and it's like an OTP, so we cant reuse it

Reply:
How many times can you use it? 

Could you then going abd get the access token, in a different script. 

Then just set the access token in an env var ready to use

Reply:
> How many times can you use it?
> 
> Could you then going abd get the access token, in a different script.
> 
> Then just set the access token in an env var ready to use

we can look at it this way, everytime you log on to the website, and want the enphase data, you need that auth code(which can only be used once) to get the access token, the access token has some validity time, which im storing in a session state, so we can keep usig that in the website

Reply:
> How many times can you use it?
> 
> Could you then going abd get the access token, in a different script.
> 
> Then just set the access token in an env var ready to use

and also this is what im doing in the website, just instead of an env var,im using session states,since env var would mean a more permanent solution(which is pointless since access tokens do expire)

Reply:
How long do they take to expire?

Reply:
> How long do they take to expire?

not quite sure, but if youre not a system owner then 1day if i rem correctly